### PR TITLE
small and large boys

### PR DESCRIPTION
**Pull request by ethteck**

### CODE REVIEW

**Comment by BowserSlug** [comment]
🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀 This PR matches 19 functions (+0.24%) on `us`.

⚠️ This PR introduces 3 warnings:
- src/165490.c:51: warning: implicit declaration of function `filemenu_draw_message'
- src/167570.c:50: warning: implicit declaration of function `filemenu_draw_message'
- src/168590.c:27: warning: implicit declaration of function `filemenu_draw_message'
- src/169BE0.c:142: warning: implicit declaration of function `filemenu_draw_message'
- src/169BE0.c:62: warning: implicit declaration of function `mainmenu_draw_rect'
- src/16F740.c:2832: warning: assignment makes pointer from integer without a cast
- src/16c8e0.c:694: warning: assignment from incompatible pointer type

**Comment by BowserSlug** [comment]
🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀 This PR matches 19 functions (+0.24%) on `us`.

File: src/16F740.c [hunk 1]
```diff
@@ -2177,12 +2176,421 @@ void btl_state_update_next_enemy(void) {
...
+                                    btl_update_ko_status();
+                                    if (player->status != 0) {
+                                        player->flags |= ACTOR_FLAG_8000000 | ACTOR_FLAG_4000000;
+                                        if (partner != NULL) {
+                                            partner->flags |= ACTOR_FLAG_8000000 | ACTOR_FLAG_4000000;
+                                        }
+                                        btl_cam_use_preset(BTL_CAM_PRESET_33);
+
+                                        switch (player->status) {
+                                            case 4:
```
Comment (nanaian) [inline src/16F740.c:2302]: Should probably use the status enum here

File: src/16F740.c [hunk 2]
```diff
@@ -2177,12 +2176,421 @@ void btl_state_update_next_enemy(void) {
...
+    if (gBattleState2 == BATTLE_STATE2_UNK_1E) {
+        btl_set_state(BATTLE_STATE_NEXT_ENEMY);
+    }
+}
 
 void btl_state_draw_enemy_move(void) {
 }
 
-INCLUDE_ASM(s32, "16F740", btl_state_update_first_strike);
+//INCLUDE_ASM(s32, "16F740", btl_state_update_first_strike);
```
Comment (nanaian) [inline src/16F740.c:2400]:
```suggestion
```

File: src/sprite.c [hunk 3]
```diff
@@ -676,13 +676,81 @@ s32 func_802DE748(s32 arg0, s32 arg1) {
...
+void func_802DE780(s32 spriteIdx, s32 arg1, s32 arg2, s32 arg3, s32 arg4, s32 arg5, s32 arg6, s32 arg7) {
+    SpriteInstance* sprite = &D_802DFA48[spriteIdx];
+    SpriteComponent** componentList;
+    s32 i;
+
+    if (sprite->componentList != NULL) {
+        componentList = sprite->componentList;
+        i = 0;
+
+        while ((s32) *componentList != -1) {
```
Comment (nanaian) [inline src/sprite.c:688]: Absurd that they didnt use null to terminate lol

File: src/world/area_mac/mac_01/805DD0.c [hunk 4]
```diff
@@ -8,5 +8,15 @@
 
 static char* N(exit_str_0) = "osr_00";
 
-INCLUDE_ASM(s32, "world/area_mac/mac_01/805DD0", func_802463E0_806C60);
-
+ApiStatus func_802463E0_806C60(Evt* script, s32 isInitialCall) {
```
Comment (nanaian) [inline src/world/area_mac/mac_01/805DD0.c:11]: Is this a dup of func_80244D68_8055E8

**Review by nanaian — CHANGES_REQUESTED** [review]

File: src/world/script_api/7E0E80.c [hunk 5]
```diff
@@ -831,13 +831,13 @@ INCLUDE_ASM(s32, "world/script_api/7E0E80", draw_shop_items);
 // This should be equivalent to the original code but there is some funny business with
 // the evt_get_variable's at the beginning that makes absolutely no sense.
```
[the remark this answers is not in the document]
Comment (ethteck) [reply src/world/script_api/7E0E80.c:832]: I didn't match it sadly. Just played with it a bit / got it closer

**Pull request merged**